java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -690,6 +690,215 @@ void testInnerJoinWithOnlyCommonKeys() {
     }
   }
 
+  @Test
+  void testLowerBound() {
+    boundsEmptyInput(false);
+
+    boundsEmptyValues(false);
+
+    boundsStringCategories(false);
+
+    boundsIntsSingleCol(false);
+
+    boundsLongsSingleCol(false);
+
+    boundsLongsMultiVal(false);
+
+    boundsMultiCol(false);
+
+    boundsUnequalCols1(false);
+
+    boundsUnequalCols2(false);
+
+    boundsNulls(false);
+  }
+
+  @Test
+  void testUpperBound() {
+    boundsEmptyInput(true);
+
+    boundsEmptyValues(true);
+
+    boundsStringCategories(true);
+
+    boundsIntsSingleCol(true);
+
+    boundsLongsSingleCol(true);
+
+    boundsLongsMultiVal(true);
+
+    boundsMultiCol(true);
+
+    boundsUnequalCols1(true);
+
+    boundsUnequalCols2(true);
+
+    boundsNulls(true);
+  }
+
+  private void boundsNulls(boolean isUpperBound) {
```

Comment:
I'd make each of these a junit test function, and have the upper/lower be a loop in each one. That is a bit more code, but it will be more clear when/if one fails in the output. Thoughts?

Reply:
Sure that would work too. When a test fails with the current patch it shows up something like this - which I thought is good enough but I am happy to change if the following output does not agree 
`Failures: 
[ERROR]   TableTest.testLowerBound:697->testBoundsEmptyInput:886->getBoundsCv:895 Input table cannot be empty
[ERROR]   TableTest.testUpperBound:722->testBoundsEmptyInput:886->getBoundsCv:894 Input table cannot be empty
`
Appreciate the feedback.

Reply:
oh that's interesting, so it automagically turns `boundEmptyInput` into `testBoundsEmptyInput`. I think that output is clearer than what you'd get otherwise with a loop inside of `testBoundsEmptyInput', so this seems good to me.